iree/compiler/Conversion/LinalgToLLVM/ConvertToLLVM.cpp

```diff
@@ -132,6 +132,34 @@ class RemoveMakeRankedShape : public ConvertToLLVMPattern {
   }
 };
 
+template <typename Op, int ArgIndex>
+class ConvertWorkgroupInfoOpPattern : public ConvertToLLVMPattern {
+ public:
+  explicit ConvertWorkgroupInfoOpPattern(MLIRContext *context,
+                                         LLVMTypeConverter &typeConverter)
+      : ConvertToLLVMPattern(Op::getOperationName(), context, typeConverter) {}
+
+  LogicalResult matchAndRewrite(
+      Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    auto newFuncOp = cast<LLVM::LLVMFuncOp>(rewriter.getBlock()->getParentOp());
+    auto xyzTy =
+        LLVM::LLVMType::getInt32Ty(rewriter.getContext()).getPointerTo();
+    auto xyzArgument = newFuncOp.getArgument(ArgIndex);
```

Comment:
Nice! this might be the only valid way to do this kind of conversion `func() -> llvm.func(%fixed_args....)`.